build/Jenkinsfile

```diff
@@ -128,23 +160,25 @@ podTemplate(label: 'jenkins-slave',
     }
     stage('Style Checks') {
         container('golang') {
-            sh 'cd /go/src/github.com/mattermost/mattermost-server && make check-style BUILD_NUMBER=$BUILD_NUMBER'
+            sh "cd /go/src/github.com/mattermost/mattermost-server && make check-style BUILD_NUMBER=${env.BUILD_NUMBER}"
         }
     }
     stage('Build') {
         container('golang') {
-            sh 'cd /go/src/github.com/mattermost/mattermost-server && make build BUILD_NUMBER=$BUILD_NUMBER'
+            sh "cd /go/src/github.com/mattermost/mattermost-server && make build BUILD_NUMBER=${env.BUILD_NUMBER}"
         }
         container('mattermost-node') {
-            sh 'cd mattermost-webapp && curl -f -o ./dist.tar.gz https://releases.mattermost.com/mattermost-webapp/commit/`git rev-parse HEAD`/mattermost-webapp.tar.gz && mkdir ./dist && tar -xvf ./dist.tar.gz -C ./dist --strip-components=1 || make build'
+            sh "cd mattermost-webapp && curl -f -o ./dist.tar.gz https://releases.mattermost.com/mattermost-webapp/commit/`git rev-parse HEAD`/mattermost-webapp.tar.gz && mkdir ./dist && tar -xvf ./dist.tar.gz -C ./dist --strip-components=1 || make build BUILD_NUMBER=${env.BUILD_NUMBER}"
         }
         container('golang') {
-            sh 'cd /go/src/github.com/mattermost/mattermost-server && make package BUILD_NUMBER=$BUILD_NUMBER'
+            sh "cd /go/src/github.com/mattermost/mattermost-server && make package BUILD_NUMBER=${env.BUILD_NUMBER}"
         }
     }
     stage('Unit Tests') {
         container('golang') {
-            sh 'cd /go/src/github.com/mattermost/mattermost-server && make test-te BUILD_NUMBER=$BUILD_NUMBER TESTFLAGS= TESTFLAGSEE='
+            withEnv(['CI_HOST=localhost', 'CI_INBUCKET_PORT=10080', 'CI_MINIO_PORT=9000']) {
```

Comment:
Can't we just use the same configuration as everyone else? CI systems doing things uniquely different than everyone else can be a real pain. When I run `make test` locally, I really want it to be as close as possible to what CI and everyone else does. That way we can avoid more breakage that "works on my machine".

Reply:
looks like not, since we are running the tests inside the golang container and in the same pod as the other containers we have access to the container port and not in the hostport. because all containers are in the same pod. I tried some variances but did not work. :(

I will keep trying to see if there is any other possible solution, but for now, this is the way

Reply:
Yeah that does make it tricky. It seems like it should be possible, but maybe not.

I guess if we have to keep the "CI_" stuff, you'll also want to update these lines:

https://github.com/mattermost/mattermost-server/blob/8e19ba029f889519d93cf272960dce858971106c/api4/apitestlib.go#L450
https://github.com/mattermost/mattermost-server/blob/8e19ba029f889519d93cf272960dce858971106c/api/apitestlib.go#L364

(Since when did GitHub make source line links so fancy? 😯)

I would probably change them to something like this:

```go
func GenerateTestEmail() string {
	if utils.Cfg.EmailSettings.SMTPServer != "dockerhost" && os.GetEnv("CI_INBUCKET_PORT") == "" {
		return strings.ToLower("success+" + model.NewId() + "@simulator.amazonses.com")
	}
	return strings.ToLower(model.NewId() + "@inbucket")
}
```

Reply:
fancy, maybe since today 😂

Reply:
done. But just a comment, I ran the pipeline and all tests passed, this change here maybe is more cosmetic 😄